libraries/bezier-rs/src/subpath/solvers.rs

```diff
@@ -10,35 +10,37 @@ impl Subpath {
 	/// Expects `t` to be within the inclusive range `[0, 1]`.
 	/// <iframe frameBorder="0" width="100%" height="400px" src="https://graphite.rs/bezier-rs-demos#subpath/evaluate/solo" title="Evaluate Demo"></iframe>
 	pub fn evaluate(&self, t: SubpathTValue) -> DVec2 {
-		let (segment_index, t) = self.t_value_to_parametric(t);
+		let (segment_index, t) = match t {
+			SubpathTValue::GlobalEuclidean(_) => todo!(),
+			SubpathTValue::Parametric { segment_index, t: segment_t } => (segment_index, segment_t),
+			SubpathTValue::GlobalParametric(_) => self.t_value_to_parametric(t),
+			SubpathTValue::Euclidean { segment_index, t } => todo!(),
+			SubpathTValue::EuclideanWithinError { segment_index, t, error } => todo!(),
+			SubpathTValue::GlobalEuclideanWithinError { t, error } => todo!(),
+		};
```

Comment:
This whole match statement can be removed

Reply:
By removed, do you mean replace all the different match branches with 1 default match branch `_ => todo!(),`? The reason I changed this assignment to a match statement is because `intersection` and `self-intersection` return `(segment_idx: usize, t_value: f64)` while other subpath functions return `GlobalParametric(t_value)`.

Reply:
I don't see why the match statement is necessary. `self.t_value_to_parametric(t)` will do the conversion from the different variants to a `(segment, index)` tuple, so there is no need to replicate within `evaluate`